### PR TITLE
cdk-java: add tests for interrupted refreshes

### PR DESCRIPTION
**Pull request by stephane-airbyte**
## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by stephane-airbyte** [comment]
* **#42514** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/42514?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stephane-airbyte and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt [hunk 1]
```diff
@@ -361,6 +363,69 @@ abstract class BaseTypingDedupingTest {
...
+                )
+
+        // First sync
+        val messages1 = readMessages("dat/sync1_messages.jsonl")
+
+        runSync(catalog1, messages1)
+
+        val expectedRawRecords1 = readRecords("dat/sync1_expectedrecords_raw.jsonl")
+        val expectedFinalRecords1 = readRecords("dat/sync1_expectedrecords_nondedup_final.jsonl")
+        verifySyncResult(expectedRawRecords1, expectedFinalRecords1, disableFinalTableComparison())
```
[the remark this answers is not in the document]
Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt:394]: yeah, you're very probably right. The @Test annotation is missing, so adding it should make that test fail

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt [hunk 2]
```diff
@@ -1027,6 +1096,156 @@ abstract class BaseTypingDedupingTest {
...
+            io.airbyte.protocol.models.v0
+                .ConfiguredAirbyteCatalog()
+                .withStreams(
+                    java.util.List.of(
+                        ConfiguredAirbyteStream()
+                            .withSyncId(42)
+                            .withGenerationId(41)
+                            .withMinimumGenerationId(41)
+                            .withSyncMode(SyncMode.FULL_REFRESH)
+                            .withCursorField(listOf("updated_at"))
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt:1111]: nit: non-dedup streams don't need a cursor/PK

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt [hunk 3]
```diff
@@ -361,6 +363,73 @@ abstract class BaseTypingDedupingTest {
...
+        val expectedRawRecords1 = readRecords("dat/sync1_expectedrecords_raw.jsonl")
+        val expectedFinalRecords1 = readRecords("dat/sync1_expectedrecords_nondedup_final.jsonl")
+        (expectedRawRecords1 + expectedFinalRecords1).forEach {
+            (it as ObjectNode).put(JavaBaseConstants.COLUMN_NAME_AB_GENERATION_ID, 0)
+        }
+        verifySyncResult(expectedRawRecords1, expectedFinalRecords1, disableFinalTableComparison())
+
+        // Second sync
+        val messages2 = readMessages("dat/sync2_messages.jsonl")
+        val catalog2 =
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt:402]: is this catalog identical to catalog1?

**Review by edgao — APPROVED** [review]

**Comment by stephane-airbyte** [comment]
### Merge activity

* **Jul 31, 5:20 PM EDT**: @stephane-airbyte started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/42514).

**Pull request merged**